tests/func/test_remote.py

```diff
@@ -215,12 +218,13 @@ def unreliable_download(_from_fs, from_info, _to_fs, to_info, **kwargs):
         for from_i, to_i in zip(from_info, to_info):
             on_error(from_i, to_i, Exception())
 
-    with patch.object(generic, "transfer", unreliable_download):
-        with pytest.raises(DownloadError) as download_error_info:
-            dvc.pull()
-        # error count should be len(.dir + standalone file checksums)
-        # since files inside dir are ignored if dir cache entry is missing
-        assert download_error_info.value.amount == 2
+    mock_download = mocker.patch.object(generic, "transfer", unreliable_download)
+    with pytest.raises(DownloadError) as download_error_info:
+        dvc.pull()
+    # error count should be len(.dir + standalone file checksums)
+    # since files inside dir are ignored if dir cache entry is missing
+    assert download_error_info.value.amount == 2
+    mocker.stop(mock_download)
```

Comment:
Definitely looked better before. But looks like that's intended so oh well https://pytest-mock.readthedocs.io/en/latest/remarks.html#usage-as-context-manager